### PR TITLE
Add 'Exit to tray' function

### PR DESCRIPTION
**Pull request by tuxzz**
### Function ###
**'Exit to tray' function let foobar2000 minimize to tray when user click 'X' button.**
1. This function makes new users feeling more natural.
2. This function gives old users one more option.

### Changes ###
1. Added 'Exit to tray' checkbox to 'Notification Area' dialog.
2. <del>Make 'Exit to tray' as default enabled and make 'Minimise to tray' as default disabled.</del>

### CODE REVIEW

File: foo_ui_columns/config_vars.cpp [hunk 1]
```diff
@@ -73,7 +73,8 @@ cfg_int cfg_back(create_guid(0x7446b27e, 0xb23d, 0xb28a, 0x05, 0x02, 0x3a, 0x2b,
     cfg_tooltips_clipped(create_guid(0x8181afae, 0x404e, 0xa880, 0xe2, 0x46, 0x38, 0x69, 0xbd, 0xc9, 0xd3, 0xef), 0),
     cfg_np(create_guid(0x2e590774, 0xc50e, 0xbcd0, 0x73, 0xa2, 0x75, 0x2f, 0x70, 0x9b, 0xb1, 0x2e), 1),
     cfg_show_systray(create_guid(0x19cf6962, 0x3d1e, 0x10b7, 0xcd, 0x91, 0x48, 0x66, 0x8b, 0x72, 0xe8, 0x22), 1),
-    cfg_minimise_to_tray(create_guid(0x4c4bf8f0, 0x1f76, 0x65f4, 0x52, 0xe4, 0x78, 0x01, 0x45, 0x13, 0xf9, 0x79), 1),
+    cfg_minimise_to_tray(create_guid(0x4c4bf8f0, 0x1f76, 0x65f4, 0x52, 0xe4, 0x78, 0x01, 0x45, 0x13, 0xf9, 0x79), 0),
+    cfg_exit_to_tray(create_guid(0x41526941, 0x4d69, 0x4b55, 0x54, 0x78, 0x5a, 0x5a, 0x57, 0x52, 0x4c, 0x44), 1),
```
Comment (reupen) [inline foo_ui_columns/config_vars.cpp]: Should be a boolean and default to off (the other ones are legacy stuff).

**Comment by tuxzz** [comment]
**Everything fixed.**

> Can the option be moved to advanced preferences as closing to the notification icon is non-standard behaviour?

I don't want to move it to advanced preferences. Closing to tray is a very common feature on other music players. For example, Amarok, Netease Music.

**Comment by reupen** [comment]
> Everything fixed.

:+1:

> I don't want to move it to advanced preferences. Closing to tray is a very common feature on other music players. For example, Amarok, Netease Music.

I'm still inclined to say it should be in advanced preferences. I don't dispute that there exist other applications with the functionality, but it's non-standard behaviour because 'close' means 'close' and not 'run in the background'. I don't want to encourage such behaviour _in any application_.

The equivalent option in the Default UI is also in advanced preferences, and the default player in Windows 10 (Groove Music) does not have this functionality. Any minimise/close-to-a-notification-icon behaviour makes less sense today when Windows shows taskbar entries without labels by default, lets you pin applications to the taskbar and tends to hide third-party notification icons. (I'd even say that the current defaults are wrong.)

**Comment by tuxzz** [comment]
**Moved to advanced preferences**

**Pull request merged**